app/models/concerns/vaccination_record_sync_to_nhs_immunisations_api_concern.rb

```diff
@@ -12,6 +12,44 @@ def syncable_to_nhs_immunisations_api?
     recorded_in_service?
   end
 
+  def sync_status
```

Comment:
I think the way we're using "synced" is a little ambiguous below. We seem to be using to indicate if a vaccination record has been sent to the Imms API, but technically "in-sync" would also mean that a record that shouldn't be in the  Imms API isn't there. The opposite can happen, for example, when a patient's NHS number is invalidated/removed., which will result in `cannot_sync` below, even while we're waiting for it to be deleted.

Also, I think we might be able to reuse `next_sync_action` for some of the logic here. Let's have a chat about this?